File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
from collections import deque
_input = sys.stdin.readline

com = int(_input())
conn = int(_input())
visited = [False] * (com + 1)
graph = [[] * (com + 1) for _ in range(com + 1)]

for _ in range(conn):
    a, b = map(int, _input().split())
    graph[a].append(b)
    graph[b].append(a)

result = 0

queue = deque([1])
visited[1] = True
while queue:
    host = queue.popleft()
    for i in range(len(graph[host])):
        if not visited[graph[host][i]]:
            queue.append(graph[host][i])
            visited[graph[host][i]] = True
            result += 1

print(result)

File: 백준/Silver/2606. 바이러스/바이러스.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		int com = Integer.parseInt(br.readLine());
		int conn = Integer.parseInt(br.readLine());
		boolean[] visited = new boolean[com + 1];
		List<Integer>[] arr = new List[com + 1];
		for (int i = 0; i <= com; i++) arr[i] = new ArrayList<Integer>();

		for (int i = 1; i <= conn; i++) {
			st = new StringTokenizer(br.readLine());
			int x = Integer.parseInt(st.nextToken());
			int y = Integer.parseInt(st.nextToken());
			arr[x].add(y);
			arr[y].add(x);
		}

		int result = 0;
		Queue<Integer> queue = new LinkedList<>();
		for (int i = 0; i < arr[1].size(); i++) {
			queue.offer(arr[1].get(i));
		}
		visited[1] = true;
		while (!queue.isEmpty()) {
			int now = queue.poll();
			if (!visited[now]) {
				visited[now] = true;
				for (int i = 0; i < arr[now].size(); i++) {
					queue.offer(arr[now].get(i));
				}
				result++;
			}
		}
		System.out.println(result);
	}
}